kernel/src/main/scala/cats/kernel/PartialOrder.scala

```diff
@@ -0,0 +1,169 @@
+package cats.kernel
+
+import scala.{specialized => sp}
+
+/**
+ * The `PartialOrder` type class is used to define a partial ordering on some type `A`.
+ * 
+ * A partial order is defined by a relation <=, which obeys the following laws:
+ * 
+ * - x <= x (reflexivity)
+ * - if x <= y and y <= x, then x = y (anti-symmetry)
+ * - if x <= y and y <= z, then x <= z (transitivity)
+ * 
+ * To compute both <= and >= at the same time, we use a Double number
+ * to encode the result of the comparisons x <= y and x >= y.
+ * The truth table is defined as follows:
+ * 
+ * x <= y    x >= y      Double
+ * true      true        = 0.0     (corresponds to x = y)
+ * false     false       = NaN     (x and y cannot be compared)
+ * true      false       = -1.0    (corresponds to x < y)
+ * false     true        = 1.0     (corresponds to x > y)
+ */
+trait PartialOrder[@sp A] extends Any with Eq[A] { self =>
+
+  /**
+   * Result of comparing `x` with `y`. Returns NaN if operands are not
+   * comparable. If operands are comparable, returns a Double whose
+   * sign is:
+   * - negative iff `x < y`
+   * - zero     iff `x = y`
+   * - positive iff `x > y`
+   */
+  def partialCompare(x: A, y: A): Double
+
+  /**
+   * Result of comparing `x` with `y`. Returns None if operands are
+   * not comparable. If operands are comparable, returns Some[Int]
+   * where the Int sign is:
+   * - negative iff `x < y`
+   * - zero     iff `x = y`
+   * - positive iff `x > y`
+   */  
+  def tryCompare(x: A, y: A): Option[Int] = {
+    val c = partialCompare(x, y)
+    if (c.isNaN) None else Some(c.signum)
```

Comment:
also, we might keep `val some1 = Some(1); val someNeg1 = Some(-1); val someZero = Some(0)` around and not allocate on each compare. These could be private in the PartialOrder object.